src/services/id-service.api-service.ts

```diff
@@ -36,4 +39,34 @@ export class IdServiceApiService {
         }
       });
   }
+
+  public async getByUri<T>(uri: string, getSso: () => Promise<string>): Promise<IHttpResponse<T>> {
+    try {
+      const sso = await getSso();
+      const response = await this.http.createRequest(`${oeAppConfig.idServiceUrl + '/uris?uri=' + uri}`)
+      .asGet()
+      .withHeader('Authorization', 'Bearer ' + sso)
+      .send() as IHttpResponse<IIdServiceResponse>; 
+
+      if (response.content.location) {
+        try {
+          const locationResponse = await this.http.createRequest(response.content.location).asGet()
+          .withHeader('Authorization', 'Bearer ' + sso)
+          .send() as IHttpResponse<T>;
+
+          const content = locationResponse.content;
+          const etag = locationResponse.headers.get('ETag');
+          return { ...locationResponse, content: { ...content, etag } };
+        } catch (e) {
+          return;
```

Comment:
analoog aan de base service class in sommige projecten kan je op regel 21 de response als exception gooien,
ofwel moet je hier weer die response.isSuccess gaan controleren